### ISSUE TITLE
Possibility to specify a base path to unpack a package

### ISSUE THREAD

**Issue by Unisay** [commercialhaskell/stack#3904]
### Feature request

Currently, `stack unpack` uses current folder to unpack a package into.

```shell
$ stack --version; and stack unpack --help                                                                                                                                  19:58:47
Version 1.6.5, Git revision 24ab0d6ff07f28276e082c3ce74dfdeb1a2ca9e9 (5514 commits) x86_64 hpack-0.20.0
Usage: stack unpack PACKAGE [--help]
  Unpack one or more packages locally
```
It would be convenient to be able to specify an optional destination path to unpack into it, for example:
```shell
$ stack unpack ekg --to /tmp                                                                                                                                                     
ekg-0.4.0.14: download
Unpacked ekg-0.4.0.14 to /tmp/ekg-0.4.0.14/
```

**Comment by mgsloan** [comment]
Hmm, I suppose I can see the convenience of this!  The alternative would be to `cd /tmp` and then `stack unpack ekg --stack.yaml /home/...`.

I'd merge a PR for this.

**Comment by Unisay** [comment]
Currently I need this as a part of a continuous delivery pipeline build where I'd like to avoid changing directory back and forth. At the moment I workaround it by redirecting stderr to stdout, parsing output folder, then moving it to a desired location.

So I am going to try creating a PR for it.

**Comment by snoyberg** [comment]
Fixed by #3947, thanks!
